src/mono/wasm/Wasm.Build.Tests/TestAppScenarios/SignalRClientTests.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Playwright;
+using Xunit.Abstractions;
+using Xunit;
+
+#nullable enable
+
+namespace Wasm.Build.Tests.TestAppScenarios;
+
+public class SignalRClientTests : AppTestBase
+{
+    public SignalRClientTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+        : base(output, buildContext)
+    {
+    }
+
+    [ConditionalTheory(typeof(BuildTestBase), nameof(IsWorkloadWithMultiThreadingForDefaultFramework))]
+    [InlineData("Debug", "LongPolling")]
+    [InlineData("Release", "LongPolling")]
+    [InlineData("Debug", "WebSockets")]
+    [InlineData("Release", "WebSockets")]
+    public async Task SignalRPassMessages(string config, string transport)
+    {
+        CopyTestAsset("BlazorHostedApp", "SignalRClientTests");
+        string frameworkDir = FindBlazorHostedBinFrameworkDir(config,
+            forPublish: false,
+            clientDirRelativeToProjectDir: "../BlazorHosted.Client");
+        BuildProject(configuration: config,
+            binFrameworkDir: frameworkDir,
+            runtimeType: RuntimeVariant.MultiThreaded);
+
+        List<string> testOutput = new();
+        List<string> consoleOutput = new();
+        List<string> serverOutput = new();
+
+        // To look for static files in the paths defined by "*staticwebassets.runtime.json"
```

Comment:
I would say more like "Since we are using build (not publish), we need to instruct static web assets to use manifest file, because wwwroot in bin doesn't contain all files (for build)" 🤷‍♂️